### PR TITLE
Keeper atomic stat

### PR DESCRIPTION
**Pull request by zhanglistar**
Changelog category (leave one):
- Performance Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Less lock on connection using atomic stat. Notice that it is an approximate stat.

> Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by alesapin** [comment]
It really give better performance? With previous version they were at least consistent, now counters will be inconsistent with each other.

**Comment by zhanglistar** [comment]
> It really give better performance? With previous version they were at least consistent, now counters will be inconsistent with each other.
new version:
Pasted Graphic 8

master version(db69ab9d17d1bbb772cc8b2c18db33df6f7502ee):
Pasted Graphic 9

File: src/Coordination/KeeperConnectionStats.cpp [hunk 1]
```diff
@@ -1,65 +1,63 @@
...
-
-    if (latency_ms > max_latency)
-    {
-        max_latency = latency_ms;
-    }
+    last_latency.store(latency_ms, std::memory_order_relaxed);
+    total_latency.fetch_add(latency_ms, std::memory_order_relaxed);
+    count.fetch_add(1, std::memory_order_relaxed);
+
+    if (latency_ms < min_latency.load(std::memory_order_relaxed))
```
Comment (alexey-milovidov) [inline src/Coordination/KeeperConnectionStats.cpp:56]: This is not correct (can lose data). Atomic min/max requires CAS.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]

**Comment by alexey-milovidov** [comment]
Everything else LGTM.

PS. I thought that requests processing is single threaded and there will be no contention on this mutex, unless stat queries are very frequent. @alesapin WDYT?

**Comment by zhanglistar** [comment]
@alexey-milovidov Thanks for your review,  KeeperTCPHandler::updateStats is in multi-thread path.

File: src/Coordination/KeeperConnectionStats.cpp (hunk 1 above)
Comment (zhanglistar) [reply src/Coordination/KeeperConnectionStats.cpp]: Yes, I am lazy. Fix now.

**Comment by alesapin** [comment]
I think it's alright.

**Review by alesapin — APPROVED** [review]

**Pull request merged**